portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,28 +29,34 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data") 
 public class DataServlet extends HttpServlet {
-
-  private ArrayList<Comment> comments = new ArrayList<Comment>(
-    Arrays.asList(new Comment("This site is awesome!!!!!!1!","Some fanboy"),
-    new Comment("Hi, I'm Andrew Yang, and I approve this message.","Andrew Yang"),
-    new Comment("Hi, I'm Andrew Ying, and I dissaprove of the above.","Andrew Ying")));
+  // declare global constants so as to not incur the risks of hard-coding strings
+  String commentEntityName = "Comment";
+  String authorFieldName = "author";
+  String commentFieldName = "comment";
+  String timestampFieldName = "timestamp";
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    //create comment and convert to Json
-    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
-    
+    // create comment and convert to Json
+    Query query = new Query(commentEntityName).addSort(timestampFieldName, SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
+    String commentsParameter = request.getParameter("num-comments");
+    System.out.println("Comments parameter is: " + commentsParameter);
+    int numComments = Integer.parseInt(commentsParameter);
+    int commentsLength = 0;
```

Comment:
If the value is initialized to zero, simply declare the variable since default for int is zero:
int commentsLength;

Reply:
I tried to do that, but it gave me the following compilation failure: "variable commentsLength might not have been initialized."